code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/machineDriver/MachineDriverUpdateActionHandler.java

```diff
@@ -0,0 +1,42 @@
+package io.cattle.platform.iaas.api.machineDriver;
+
+import io.cattle.platform.api.action.ActionHandler;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.model.tables.records.MachineDriverRecord;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class MachineDriverUpdateActionHandler implements ActionHandler{
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public Object perform(String name, Object obj, ApiRequest request) {
+        if (obj instanceof MachineDriverRecord) {
```

Comment:
Don't do instanceof on or cast to the implementation (MachineDriverRecord), use the interface (MachineDriver).